### ISSUE TITLE
'no mix space and tab' can not fix this code

### ISSUE THREAD

**Issue by craigcosmo** [eslint/eslint#8178]
* **ESLint Version:**
3.16.1
* **Node Version:**
6
* **npm Version:**
3.10.10

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**
```json
{
	"parser": "babel-eslint",
	"parserOptions": {
		"ecmaFeatures": {
			"jsx": true,
			"modules": true
		}
	},
	"env": {
		"browser": true,
		"es6": true,
		"node": true,
		"jquery":true,
		"jasmine":true,
		"mocha":true,
		"amd":true
	},
	"plugins": [
		"react",
	],
	"rules": {
		"quote-props": ["off", "as-needed"],
		"no-unused-vars": ["off"],
		"indent": ["warn", "tab", { "SwitchCase": 1 }],
		"no-mixed-spaces-and-tabs": "warn",
		"jsx-quotes": ["warn", "prefer-double"],
		"quotes": ["warn", "single"],
		"comma-dangle": ["warn", "only-multiline"],
		"no-unneeded-ternary": "off",
		"react/jsx-indent": ["warn", "tab"],
		"react/jsx-indent-props": ["warn", "tab"]
	}
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
export default class Home extends React.Component {
	constructor(props) {
		super(props)
	 }
	 componentDidMount(){
	 	
	}
}
```
<img width="454" alt="screen shot 2017-03-02 at 2 11 59 pm" src="https://cloud.githubusercontent.com/assets/538657/23496770/7ce148a6-ff52-11e6-8da0-06a7cf0c917a.png">

**What did you expect to happen?**

`--fix` command should fix this case

**What actually happened? Please include the actual, raw output from ESLint.**

eslint detects the error but `--fix` command did not fix this case

<img width="571" alt="screen shot 2017-03-02 at 2 14 47 pm" src="https://cloud.githubusercontent.com/assets/538657/23496807/a7c90892-ff52-11e6-822b-d0a6c0ac4179.png">

**Comment by platinumazure** [comment]
Not all rules are fixable even when using `--fix`. Using that flag simply means all rules which have fix logic implemented will run their logic.

In the case of `no-mixed-spaces-and-tabs`, it's not clear whether the rule should fix to tabs or spaces. Rules are designed not to know about each other, so it wouldn't be acceptable for the rule to look at your `indent` configuration. Maybe the rule could be enhanced to accept a configuration option to guide it in autofixing, but at the moment, it simply doesn't have fix logic.

**Comment by craigcosmo** [comment]
You are right, I thought this rule is fixable, because it indeed look like it's fixable at first glance. I think it is a valid idea to implement fixability for it.

**Comment by not-an-aardvark** [comment]
Closing this issue since the question seems to be answered.
